packages/browser-repl/src/components/utils/inspect.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const utilInspect = require('util').inspect;
+const utilInspectCustom = utilInspect.custom || 'inspect';
+
+const formatBsonType = (value): any => ({
+  [utilInspectCustom](): string {
+    return `${value._bsontype}(${(JSON.stringify(value))})`;
+  }
+});
+
+function isBsonType(value): any {
+  return value && value._bsontype;
+}
+
+function isObject(value): any {
+  return value && typeof value === 'object' && !Array.isArray(value);
+}
+
+function formatProperty(value): any {
+  if (isObject(value) && isBsonType(value)) {
+    return formatBsonType(value);
+  }
+
+  return value;
+}
+
+function formatObject(object): any {
+  const viewObject = {};
+  for (const key of Object.keys(object)) {
+    viewObject[key] = formatProperty(object[key]);
+  }
+  return viewObject;
+}
+
+function toViewValue(value): any {
+  if (isBsonType(value)) {
```

Comment:
similar comment to the above: js-bson package _might_ be helpful here to not have these extra checks.